src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -393,13 +393,33 @@ internal int Compare(ReadOnlySpan<char> string1, string string2, CompareOptions
 
         internal virtual int CompareOptionNone(ReadOnlySpan<char> string1, ReadOnlySpan<char> string2)
         {
+            // Check for empty span or span from a null string
+            if (string1.IsEmpty)
```

Comment:
You can just do this as:

```
                    if (span.Length == 0 || string2.Length == 0)
                        return string1.Length - string2.Length;
```

It is how the same check is done in MemoryExtensions.Fast.cs.